pkg/controller/utils/utils.go

```diff
@@ -274,62 +271,32 @@ func GetAmazonCloudIntegration(ctx context.Context, client client.Client) (*oper
 	return instance, nil
 }
 
-// getAuthentication
-func GetAuthentication(ctx context.Context, client client.Client) (interface{}, error) {
-	var authenticationConfig interface{}
-
-	authenticationCR := &operatorv1.Authentication{}
-	if err := client.Get(ctx, DefaultTSEEInstanceKey, authenticationCR); err != nil {
-		if !kerrors.IsNotFound(err) {
-			return nil, err
+// GetAuthentication finds the authentication CR in your cluster. If you do not want to return an error, but (nil,nil)
```

Comment:
Why have you change this, and removed the type checking through the project? I did it this way so that this would be extendable to more than just OIDC. There could be custom configuration that needs to be calculated that's not in the CR, which is why I created a new config object to hold that for OIDC i.e. like the stuff you look up from the jwksurl. There could be more stuff like this for other authentication types, so the idea is that you just return the configuration struct for the specific type, pass that single configuration into the render method (instead of requiring the function definition to hold all types), and then just switch on the type for the parts you need to configure.

Reply:
All the kibana settings other than the clientSecret are now static.